src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,104 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    auto transpose = as_type_ptr<op::v1::Transpose>(n);
+
+    PartialShape shape = n->input_value(0).get_partial_shape();
+    int dyn_dim_index = 0;
+    bool is_dyn_dim = false;
+    if (shape.is_dynamic())
+    {
+        if (shape.rank().is_dynamic())
+        {
+            return false;
+        }
+
+        int count_dynamic_dims = 0;
+        for (int i = 0; i < shape.rank().get_length(); i++)
+        {
+            if (shape[i].is_dynamic())
+            {
+                count_dynamic_dims++;
+                shape[i] = 2; // any non 1 number
+                dyn_dim_index = i;
+            }
+        }
+
+        if (count_dynamic_dims > 1)
+        {
+            return false; // only works on one dynamic dim
```

Comment:
Why it works for only one dyn dim?

Reply:
In OV we usually get ShapeOf of that node, than use Split(num_splits=rank) and than Concatenate it in the order we like with additional dimensions equal to one. Does it seem sane?